File: export_ziwei.js

const fs = require('fs');
const content = fs.readFileSync('assets/js/ziwei_data_P.js', 'utf-8');
// Remove const declaration and semicolon
// We expect: const ZIWEI_DATA_P = { ... };
// We want: { ... }
let jsonContent = content.replace(/const\s+ZIWEI_DATA_P\s*=\s*/, '');
// Remove trailing semicolon if present
jsonContent = jsonContent.trim();
if (jsonContent.endsWith(';')) {
    jsonContent = jsonContent.slice(0, -1);
}

// Eval it? Or simpler: vm.runInNewContext?
// Using eval is easiest for simple object
// However, if the file is large, eval might be fine.
// But wait, if keys are not quoted (valid JS but not JSON), eval handles it.
// If it is strict JSON, JSON.parse works.
// Given previous view, keys ARE quoted.
// Try JSON.parse first. If fails, use eval.

try {
    const data = JSON.parse(jsonContent);
    console.log(JSON.stringify(data, null, 2));
} catch (e) {
    // Fallback to eval
    // Note: eval in strict mode?
    // We can use Function
    try {
        const data = new Function('return ' + jsonContent)();
        console.log(JSON.stringify(data, null, 2));
    } catch (e2) {
        console.error("Error parsing JS:", e2);
        process.exit(1);
    }
}

File: export_ziwei.py

import json
import re

with open('assets/js/ziwei_data_P.js', 'r', encoding='utf-8') as f:
    content = f.read()

# Strip const ... = 
content = re.sub(r'const\s+ZIWEI_DATA_P\s*=\s*', '', content)
# Strip trailing ;
content = content.strip()
if content.endswith(';'):
    content = content[:-1]

try:
    data = json.loads(content)
    # Success
    with open('existing_data.json', 'w', encoding='utf-8') as f:
        json.dump(data, f, ensure_ascii=False, indent=2)
    print("Success")
except json.JSONDecodeError as e:
    print(f"Error decoding JSON: {e}")
    # Plan B: ast.literal_eval might work if it's python-like dict syntax?
    # JS object syntax is close to Python dict, but true/false vs True/False, null vs None.
    # We can try to replace tokens.
    
    # Try simple token replacements
    content = content.replace('true', 'True').replace('false', 'False').replace('null', 'None')
    try:
        import ast
        data = ast.literal_eval(content)
        with open('existing_data.json', 'w', encoding='utf-8') as f:
             json.dump(data, f, ensure_ascii=False, indent=2)
        print("Success with ast")
    except Exception as e2:
        print(f"Error with ast: {e2}")

